docs/architecture.md

```diff
@@ -0,0 +1,130 @@
+   (hub cluster, fleet-member-\* namespaces)
+
+   Internal mirror of MemberCluster resource, hosted in fleet-member-\* namespaces for communication with member clusters.  Owned by the
+   (hub) membercluster controller.
+
+### Cluster resource placement (CRP)
+
+The following resources are defined for the Cluster resource placement (CRP) use case:
+
+1. [ClusterResourcePlacement.fleet.azure.com](https://github.com/Azure/fleet/blob/main/apis/v1alpha1/clusterresourceplacement_types.go#:~:text=type%20ClusterResourcePlacement%20struct%20%7B)
+   (hub cluster, end user namespaces)
+
+   Lists names of cluster (non-namespaced) resources (e.g. *ClusterRoles*, *Namespaces*, etc.) to replicate from hub to member(s), as well
+   as the selection of member(s) to which to replicate.  Replicating a namespace implies replicating all the contained resources as well.
+   Owned by the end user.
+
+1. [Work.multicluster.x-k8s.io](https://github.com/Azure/k8s-work-api/blob/master/pkg/apis/v1alpha1/work_types.go#:~:text=type%20Work%20struct%20%7B)
+   (hub cluster, fleet-member-\* namespaces)
+
+   Lists resource specifications that the member cluster should apply.  Owned by the (hub) clusterresourceplacement controller.
+
+1. [AppliedWork.multicluster.x-k8s.io](https://github.com/Azure/k8s-work-api/blob/master/pkg/apis/v1alpha1/appliedwork_type.go#:~:text=type%20AppliedWork%20struct%20%7B)
+   (member cluster, not namespaced)
+
+   Lists metadata of resources applied locally on member cluster.  Owned by the (member) work controller.
+
+## Controllers
+
+### Join/Leave
+
+The following controllers are implemented for the Join/Leave use case:
+
+1. [pkg/controllers/membercluster](https://github.com/Azure/fleet/tree/main/pkg/controllers/membercluster) (hub cluster)
+
+   * Watches *MemberClusters*, ensures fleet-member-\* *Namespace*, *Role*, *RoleBinding* and *InternalMemberCluster* resources.
+   * Copies *InternalMemberCluster* status to *MemberCluster* status.
+
+1. [pkg/controllers/internalmembercluster](https://github.com/Azure/fleet/tree/main/pkg/controllers/internalmembercluster) (member cluster)
+
+   * Watches *InternalMemberCluster* on hub, ensures work controller running, updates heartbeat, sets resource and health status.
+
+### Cluster resource placement (CRP)
+
+Cluster resource placement (CRP) uses a shared [pkg/resourcewatcher](https://github.com/Azure/fleet/tree/main/pkg/resourcewatcher) component
+to watch the hub for changes on *all* resources and enqueue changes of interest to different controllers as follows:
+
+* Create/update/delete of *ClusterResourcePlacements* -> clusterresourceplacement controller.
+* Update/delete of *Works* -> clusterresourceplacement controller.
+* Update of *MemberClusters* -> memberclusterplacement controller.
+* Create/update/delete of any[*] watchable resource -> resourcechange controller.
```

Comment:
It might be good to explain what the resourceChange controller does to make the picture whole. There is an internal architect diagram you can borrow too.